app/models/tag.rb

```diff
@@ -54,8 +53,17 @@ def taggings_count_cache_key
   end
 
   def write_taggings_to_redis(value)
+    # Atomically set the value while extracting the old value.
+    old_redis_value = REDIS_GENERAL.getset("tag_update_#{id}_value", value)
+
+    # If the value hasn't changed from the saved version or the REDIS version,
+    # there's no need to write an update to the database, so let's just bail
+    # out.
+    return value if value == old_redis_value && value == taggings_count_cache
+
+    # If we've reached here, then the value has changed, and we need to make
+    # sure that the new value is written to the database.
     REDIS_GENERAL.sadd("tag_update", id)
```

Comment:
I think the old code does this already:

https://redis.io/commands/sadd

"Add the specified members to the set stored at key. Specified members that are already a member of this set are ignored."

And treating it as a set means a maximum of one call, rather than a maximum of two.

Reply:
It does sometimes incorrectly report _other_ numbers, including negatives, so it certainly seems plausible it would incorrectly report zero as well. I agree that better save that sorry is a good approach when deleting anything.